commons/zenoh-codec/src/core/zint.rs

```diff
@@ -112,14 +112,24 @@ where
     fn write(self, writer: &mut W, mut x: u64) -> Self::Output {
         writer.with_slot(VLE_LEN, move |buffer| {
             let mut len = 0;
-            let mut b = x as u8;
-            while x > 0x7f {
-                buffer[len] = b | 0x80;
+            while (x & !0x7f) != 0 {
```

Comment:
I would suggest to specify explicitly the type of 0x7f literal as 0x7fu64.

Reply:
Done